nameko/messaging.py

```diff
@@ -105,30 +101,36 @@ def spam(self, data):
         self.exchange = exchange
         self.queue = queue
 
-    # TODO: should be a module level function
-    def get_connection(self):
-        conn = Connection(self.amqp_uri)
-        return connections[conn].acquire(block=True)
-
-    # TODO: should be a module level function
-    def get_producer(self):
-        conn = Connection(self.amqp_uri)
-        return producers[conn].acquire(block=True)
-
-    def setup(self):
+    @property
+    def amqp_uri(self):
+        return self.container.config[AMQP_URI_CONFIG_KEY]
 
-        self.amqp_uri = self.container.config[AMQP_URI_CONFIG_KEY]
+    @property
```

Comment:
don't feel that strongly. python tends to take the "we're all consenting adults" approach as opposed to making things hard to change. not sure if this is something to worry about in practice. i guess regular attrs is a bit less code, but don't mind